week1.md

```diff
@@ -0,0 +1,114 @@
+
+Our staff did a good job of assessing student understanding in the moment, and we made the call to spend a good chunk of Friday morning talking about forms as well. This means we didn't have nearly as much project time as we would have liked, but it did give us the opportunity to double down on event handling and callbacks, making the connection both to previous event handling in React and to JavaScript fundamentals.
+
+## Practice, Practice, Practice
+
+One of the things we tried with React was using a sequence of small, highly-scaffolded exercises instead of one big project for the week. We follow a similar pattern when teaching a new programming language, but this is the first time we've done it to introduce a framework.
+
+Here is the general strategy:
+- Plenty of worksheets and exercises throughout the week for immediate reinforcement
+- One small, targeted project each day, focused on whatever we just learned
+- Projects should be highly scaffolded, especially in the beginning
+  - Start with "fill in the render functions on these two components"
+  - Move towards "here is a list of components to build"
+
+As an example, take the [Litter Patrol](https://github.com/Adagold/litter-patrol) assignment from day 3. Most of the structure of the app is already there, but the students must implement state management (points, litter locations), and have to use callbacks to respond to user events in a child component. The project is easy enough to be done in an afternoon but still feels like a complete app.
+
+![Litter Patrol](litter-patrol.gif) _Completed Litter Patrol project_
+
+In practice, this technique seems to have worked very well. We have seen strong retention of the material that feels deeper than rote memorization. One of the big advantages is that it allows us to model high-level project organization, so that even on the last more open-ended project we spent much less time debugging tangled messes than we expected to.
+
+The downside of this strategy is that it's a lot of work for the instructional team, both up-front (building and maintaining multiple worksheets, exercises and projects) and in class (keeping track of what's assigned / due when).
+
+To help offset the up-front cost, we've partnered with the bootcamp [Code Platoon](www.codeplatoon.org). They gave us access to some of their React projects, and in exchange we'll be helping them transition their curriculum to Python over the next few months. Having tried and true assignments to build on made running this curriculum for the first time much smoother. Litter Patrol is an adaptation of their Whack-a-Mole project.
+
+## Call Me by Your Name
+
+The next lesson is one that took us by surprise but really shouldn't have: [clear naming conventions](https://en.wikipedia.org/wiki/Naming_convention_%28programming%29) are important. In particular, we needed to name things in a way that
```

Comment:
Just a note that we should have this as a part of our textbook curriculum as well.